src/threema/container.ts

```diff
@@ -567,16 +567,31 @@ class Messages implements threema.Container.Messages {
      * Messages must be sorted ascending by date.
      */
     public addNewer(receiver: threema.BaseReceiver, messages: threema.Message[]): void {
+        // Ignore empty list
         if (messages.length === 0) {
-            // do nothing
             return;
         }
         const receiverMessages = this.getReceiverMessages(receiver);
-        // if the list is empty, add the current message as ref
+
+        // Set oldest message if none previously available
         if (receiverMessages.list.length === 0) {
             receiverMessages.referenceMsgId = messages[0].id;
         }
-        receiverMessages.list.push.apply(receiverMessages.list, messages);
+
+        // Append new messages
+        for (const message of messages) {
+            // Determine sort key (if not set)
+            // Note: This resolves an ugly interface violation of temporary
+            //       messages.
+            if (message.sortKey === undefined) {
+                message.sortKey = receiverMessages.list.length > 0
+                    ? receiverMessages.list[receiverMessages.list.length - 1].sortKey
+                    : 0;
```

Comment:
Note that in theory the sort key of an actual message could be negative, since the field is just an integer.

Maybe `Number.MIN_SAFE_INTEGER`?

Alternatively update the ARP and state that values may not be negative.

Reply:
The latter makes more sense to me.

Reply:
https://github.com/threema-ch/app-remote-protocol/pull/6